controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,20 +125,432 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
 	return result, nil
 }
 
+func (r *StorageClassClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&ocsv1alpha1.StorageClassClaim{}, builder.WithPredicates(
+			predicate.GenerationChangedPredicate{},
+		)).
+		Owns(&rookCephv1.CephBlockPool{}).
+		Owns(&rookCephv1.CephFilesystemSubVolumeGroup{}).
+		Owns(&rookCephv1.CephClient{}).
+		Complete(r)
+}
+
 func (r *StorageClassClaimReconciler) reconcileConsumerPhases(ctx context.Context, storagecluster *v1.StorageCluster) (reconcile.Result, error) {
 	log := ctrllog.FromContext(ctx)
 	log.Info("Running StorageClassClaim controller in Consumer Mode")
 	return reconcile.Result{}, nil
 }
 
-func (r *StorageClassClaimReconciler) reconcileProviderPhases(ctx context.Context, storagecluster *v1.StorageCluster) (reconcile.Result, error) {
-	log := ctrllog.FromContext(ctx)
-	log.Info("Running StorageClassClaim controller in Converged/Provider Mode")
+func (r *StorageClassClaimReconciler) reconcileProviderPhases() (reconcile.Result, error) {
+	r.log.Info("Running StorageClassClaim controller in Converged/Provider Mode")
+
+	r.storageClassClaim.Status.Phase = string(v1alpha1.StorageClassClaimStateInitializing)
+
+	storageConsumerList := &ocsv1alpha1.StorageConsumerList{}
+	err := r.Client.List(r.ctx, storageConsumerList, client.InNamespace(r.namespace))
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to list storage consumer list. %v", err)
+	}
+
+	r.storageConsumer = &ocsv1alpha1.StorageConsumer{}
+	r.storageConsumer.Namespace = r.namespace
+	for _, consumer := range storageConsumerList.Items {
+		if r.storageClassClaim.Labels[controllers.ConsumerUUIDLabel] == string(consumer.GetUID()) {
+			r.storageConsumer.Name = consumer.Name
+		}
+	}
+
+	if err = r.get(r.storageConsumer); err != nil {
+		return reconcile.Result{}, err
+	}
```

Comment:
Why are we listing all storage consumers and then filtering by label on the operator side? this operation will become very costly as the number of storage consumers will grow. 
Also, why are we then running a get operation even after we already loaded that exact CT using the previous list operation?

There are 2 better options:

1. list with a label selector which will push the filtering into the etcd side and the response to the pod will only include a list with a single item. (and please use the `r.list` util to do so) and then save it in the `r.storageConsumer` field.

2. Just read the name of the storage consumer (which is what you are looking for) from the owner reference in the storage class claim CR and then use a `r.get` operation to load the storage consumer cr.

I would prefer option 2 as it will eliminate any kind of listing and filtering even inside the k8s side.

Reply:
ack, going with 2 option